addon/commands/_private/rdfa-commands/wrap-resource.ts

```diff
@@ -0,0 +1,67 @@
+import { addProperty } from './add-property';
+import { Command } from 'prosemirror-state';
+import { wrapIn } from 'prosemirror-commands';
+import { v4 as uuidv4 } from 'uuid';
+import {
+  findNodeByRdfaId,
+  getChildLiterals,
+} from '@lblod/ember-rdfa-editor/utils/rdfa-utils';
+
+export function wrapResource({ uriBase }: { uriBase: string }): Command {
+  return (state, dispatch) => {
+    const attrs = {
+      __rdfaId: 'placeholder',
+      rdfaNodeType: 'resource',
+      resource: 'another placeholder',
+    };
+    const wrapArgs: Parameters<typeof wrapIn> = [
+      state.schema.nodes.block_rdfa,
+      attrs,
+    ];
+    if (!wrapIn(...wrapArgs)(state)) {
+      return false;
+    }
+    if (dispatch) {
+      const objectId = uuidv4();
```

Comment:
very useful to have the editor mint a new URI for you, but there will be cases where we will want to explicitly provide a pre-existing URI (e.g. when referring to external uris)

the command could perhaps take in a flag/optional arg to allow this (the UI also needs to have a toggle)

I'd wrap the URI generating logic in a util as well to make sure we use the same logic everywhere (even though it's only two lines)

If you'd prefer, you can do this in a separate PR, I'll leave that up to you

Reply:
I'll do it in this one.